src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsReadStream.cs

```diff
@@ -0,0 +1,340 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A stream which handles reading TDS messages from the wire, and extracting the data from the 
+    /// packets in the message packets.
+    /// </summary>
+    internal class TdsReadStream : Stream, ITdsReadStream
+    {
+        // TODO: Handle Cancellation tokens in all async paths.
+        #region Private Fields
+
+        private Stream _underlyingStream;
+
+        // The buffer to hold the TDS read data.
+        private byte[] _readBuffer;
+
+        // The read pointer inside the buffer.
+        private int _readIndex = 0;
```

Comment:
nit: Initializing to the default value is redundant
nit: Either `//` comments, `///` comments, or no comments? I'd be ok with just `//` comments for private fields.

Reply:
I would like to keep the initialization. While dealing with so many numbers and offsets, the initialization, though redundant helps understand code better.
In the current implementation of the SqlClient, I always get lost in what is initialized, what is default and where.

Fixed the comments style